File: oa-portal/problems/c23-range-xor-queries/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    static int[] rangeXors(int[] x, int[] a, int[] b) {
        // write your code here

        return new int[a.length];
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt(), q = in.nextInt();
        int[] x = new int[n + 1];
        for (int i = 1; i <= n; i++) x[i] = in.nextInt();
        int[] a = new int[q], b = new int[q];
        for (int i = 0; i < q; i++) { a[i] = in.nextInt(); b[i] = in.nextInt(); }

        StringBuilder sb = new StringBuilder();
        for (int v : rangeXors(x, a, b)) sb.append(v).append('\n');
        System.out.print(sb);
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}

File: oa-portal/problems/c23-range-xor-queries/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> rangeXors(const vector<int>& x, const vector<pair<int, int>>& queries) {
    // write your code here

    return vector<int>(queries.size(), 0);
}

int main() {
    int n, q;
    if (scanf("%d %d", &n, &q) != 2) return 0;
    vector<int> x(n + 1, 0);
    for (int i = 1; i <= n; i++) scanf("%d", &x[i]);
    vector<pair<int, int>> queries(q);
    for (auto &qr : queries) scanf("%d %d", &qr.first, &qr.second);

    string out;
    for (int v : rangeXors(x, queries)) { out += to_string(v); out += '\n'; }
    fwrite(out.data(), 1, out.size(), stdout);
    return 0;
}
